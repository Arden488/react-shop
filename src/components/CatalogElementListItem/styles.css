.product-item:hover {
  box-shadow: 0 0 10px rgba(0, 0, 0, .15);
}

.product-item__image {
  text-align: center;
}

.product-item__image img {
  max-width: 100%;
}

.product-item__link {
  display: block;
  padding: 20px;
  color: black;
  text-decoration: none;
}

.product-item__title {
  display: block;
  margin: 10px 0;
}

.product-item__controls {
  display: grid;
  grid-auto-flow: column;
  justify-content: space-between;
}
