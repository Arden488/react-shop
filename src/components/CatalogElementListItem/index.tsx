import * as React from 'react';
import { Link } from 'react-router-dom';

import './styles.css';

interface ICatalogElementListItemProps {
  id: number,
  key: number,
  title: string,
  price: string,
  image: string,
}

const CatalogElementListItem: React.SFC<ICatalogElementListItemProps> = props => (
  <div className="product-item">
    <Link className="product-item__link" to={`/product/${props.id}`}>
      <div className="product-item__image"><img src={`/img/${props.image}`} alt={props.title} /></div>
      <div className="product-item__title">{props.title}</div>
      <div className="product-item__controls">
        <div className="product-item__price">{props.price} ₽</div>
        <div className="product-item__buy">
          <button>Купить</button>
        </div>
      </div>
    </Link>
  </div>
);

export default CatalogElementListItem;
