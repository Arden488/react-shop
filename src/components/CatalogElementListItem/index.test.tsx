import { shallow } from 'enzyme';
import * as React from 'react';

import CatalogElementListItem from './index';

let component: React.ReactNode = null;

beforeEach(() => {
  component = shallow(<CatalogElementListItem code="1" />);
});

it('renders correctly', () => {
  expect(component).toMatchSnapshot();
});
