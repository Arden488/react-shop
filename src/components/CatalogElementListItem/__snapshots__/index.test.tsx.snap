// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="product-item"
>
  <Link
    className="product-item__link"
    replace={false}
    to="/product/1"
  >
    <div
      className="product-item__image"
    >
      <img
        alt="Apple iPhone 8 32 ГБ (PRODUCT)RED"
        src="/img/iphone8_red.webp"
      />
    </div>
    <div
      className="product-item__title"
    >
      Apple iPhone 8 32 ГБ (PRODUCT)RED
    </div>
    <div
      className="product-item__controls"
    >
      <div
        className="product-item__price"
      >
        39990
         ₽
      </div>
      <div
        className="product-item__buy"
      >
        <button>
          Купить
        </button>
      </div>
    </div>
  </Link>
</div>
`;
